source/opt/instrument_pass.cpp

```diff
@@ -298,19 +311,55 @@ uint32_t InstrumentPass::GetOutputBufferBinding() {
   return 0;
 }
 
+uint32_t InstrumentPass::GetInputBufferBinding() {
+  switch (validation_id_) {
+    case kInstValidationIdBindless:
+      return kDebugInputBindingBindless;
+    default:
+      assert(false && "unexpected validation id");
+  }
+  return 0;
+}
+
+analysis::Type* InstrumentPass::GetUintRuntimeArrayType(
+    analysis::DecorationManager* deco_mgr, analysis::TypeManager* type_mgr) {
+  if (uint_rarr_ty_ == nullptr) {
+    analysis::Integer uint_ty(32, false);
+    analysis::Type* reg_uint_ty = type_mgr->GetRegisteredType(&uint_ty);
+    analysis::RuntimeArray uint_rarr_ty_tmp(reg_uint_ty);
+    uint_rarr_ty_ = type_mgr->GetRegisteredType(&uint_rarr_ty_tmp);
+    uint32_t uint_arr_ty_id = type_mgr->GetTypeInstruction(uint_rarr_ty_);
+    deco_mgr->AddDecorationVal(uint_arr_ty_id, SpvDecorationArrayStride, 4u);
```

Comment:
What happens if an OpRuntimeArray of uints already exists?  What if the existing on already has an array stride of 8?  We might want to test for that.

Reply:
Good question. I did think about this. I actually tried to add the decorations to the type here but the current interface did not allow me to do it and so I decided, partly for expediency, partly out of lack of confidence in my understanding of the TypeManager, to go this route and just invalidate the TypeManager at the end.

I agree I probably should go back and fix the TypeManager so I can do it this way. But I do think what I am doing is not creating the possibility of invalid code. This is because when I create an *undecorated* RuntimeArray of uint32, I am not going to get a pre-existing RuntimeArray because any pre-existing RuntimeArray of uint32 will be *decorated*. Since I will not be getting a pre-existing type, I am free to decorate it.

I agree this logic is pretty esoteric. If you are open to it, I could document this a little better. If you are not open to it, I can try to wrestle the TypeManager into submission.

Reply:
You can document it, and leave it as is.  Reference the spec when pointing out that each array type has to have a stride.